### PR TITLE
Moved the character validation to a separate class to be further expa…

### PR DESCRIPTION
**Pull request by davidgarry1**
Moved the character validation to a separate class to be further expanded for other types of text validation.

### CODE REVIEW

File: src/codeu/chat/client/simplegui/UserPanel.java [hunk 1]
```diff
@@ -163,17 +176,14 @@ public void actionPerformed(ActionEvent e) {
...
-          if(s.matches("[a-zA-Z0-9]+")) {
-            clientContext.user.addUser(s);
-            UserPanel.this.getAllUsers(listModel);
-          } else {
-            JOptionPane.showMessageDialog(UserPanel.this,
-                "User not created. Alphanumeric characters only, with no spaces. Please try again.",
-                "User Not Created",
-                JOptionPane.ERROR_MESSAGE);
-          }
+        if (validator.isValidUserName(s)) {
```
Comment (shuai9532) [inline src/codeu/chat/client/simplegui/UserPanel.java:179]: can we just call `isVaidName()` in clientUser?  I am not sure which one will be more efficient. It seems to me that we compile that pattern once again by calling `TextValidator` constructor while in `ClientUser`, we have called it once.
we can do:
```
ClientUser usr = clientContext.user;
if( ! usr.isValidName(s) ){
    ....
}
else{
    usr.addUser(s);
}
```
this way, for sure we only trigger `TextValidator` once. What do you think?

File: src/codeu/chat/util/TextValidator.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * Class used to determine whether or not certain inputs are valid for different fields throughout
+ * the program
+ */
+public final class TextValidator {
+
+  private static final Pattern validUserNamePattern = Pattern.compile("[a-zA-Z0-9]+");
+
```
Comment (shuai9532) [inline src/codeu/chat/util/TextValidator.java:12]: write method header here 
```
/**
 * @param .....
 * @return .....
 */
```

**Review by shuai9532 — CHANGES_REQUESTED** [review]
great work. I like the new username validation.
I wrote some comments and might(might not) need to change some code

File: src/codeu/chat/client/simplegui/UserPanel.java (hunk 1 above)
Comment (davidgarry1) [reply src/codeu/chat/client/simplegui/UserPanel.java:179]: I mean we are talking about constants so small here it really doesn't matter. This program has other features that are performing in worse than linear time, this function will make absolutely 0 difference in the runtime of the program. It's more about functionality, clarity, and maintainability

File: src/codeu/chat/util/TextValidator.java (hunk 2 above)
Comment (davidgarry1) [reply src/codeu/chat/util/TextValidator.java:12]: Will do

**Pull request closed without merging**